debezium-core/src/main/java/io/debezium/relational/TableId.java

```diff
@@ -36,11 +32,14 @@ public static TableId parse(String str) {
      * @param useCatalogBeforeSchema {@code true} if the parsed string contains only 2 items and the first should be used as
      *            the catalog and the second as the table name, or {@code false} if the first should be used as the schema and the
      *            second as the table name
+     * @param identifierTransformer transformer used for unquoting the identifier parts in the given string
      * @return the table ID, or null if it could not be parsed
      */
-    public static TableId parse(String str, boolean useCatalogBeforeSchema) {
-        String[] parts = IDENTIFIER_SEPARATOR.split(str);
-        if ( parts.length < 0 ) return null;
+    public static TableId parse(String str, boolean useCatalogBeforeSchema, IdentifierTransformer identifierTransformer) {
```

Comment:
I think it's better to keep things simple, and to support all the quote styles for the DBMSes we support without having to specify the parameterization. We can always rework it later if we need to.